hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -77,9 +77,11 @@ public void commitMigration(PartitionMigrationEvent event) {
 
         if (SOURCE == event.getMigrationEndpoint()) {
             clearMapsHavingLesserBackupCountThan(event.getPartitionId(), event.getNewReplicaIndex());
-            getMetaDataGenerator().resetMetadata(event.getPartitionId());
+            getMetaDataGenerator().removeUuidAndSequence(event.getPartitionId());
         } else if (DESTINATION == event.getMigrationEndpoint()) {
-            getMetaDataGenerator().getOrCreateUuid(event.getPartitionId());
+            if (event.getNewReplicaIndex() != 0) {
```

Comment:
new puts will create uuid on promoted partition, it shouldn't be an issue for this reason.